tutorials/gui_for_python_applications/README.md

```diff
@@ -0,0 +1,187 @@
+
+This tutorial demonstrates how GUI controls were integrated into the [Florence-2](https://github.com/nvidia-holoscan/holohub/tree/main/applications/florence-2-vision) Python application using [PySide6](https://doc.qt.io/qtforpython-6/). This addition enables users to dynamically change the vision task performed by the application.
+
+<p align="center">
+  <img src="./demo.gif" alt="Holoscan VILA Live">
+</p>
+
+## Overview
+
+The Florence-2 application includes the typical components of a Holohub application, with the addition of a GUI component. 
+The main components are:
+
+- <b>Dockerfile</b>: For installing additional dependencies.
+- <b>Application Code</b>: Defines the Holoscan application and its operators.
+- <b>GUI Code</b>: Utilizes PySide6 to add UI controls.
+
+
+### Dockerfile
+
+The Dockerfile is used in Holohub when the application requires additional dependencies. For GUI functionality, this application's 
+[Dockerfile](https://github.com/nvidia-holoscan/holohub/blob/main/applications/florence-2-vision/Dockerfile) installs:
+
+* `qt6-base-dev` for Qt6 framework
+* `PySide6` for Python bindings for Qt6 (as specified in [requirements.txt](https://github.com/nvidia-holoscan/holohub/blob/main/applications/florence-2-vision/requirements.txt))
+
+### Application Code
+
+The Florence-2 application code is organized across several files:
+
+* `florence2_app.py`: Main application code.
+* `florence2_op.py`: Florence-2 model inference code.
+* `florence2_postprocessor_op.py`: Post-processing code to send overlays (e.g., bounding boxes, labels, segmentation masks) to Holoviz.
+* `config.yaml`: Default application parameters.
+
+The Florence-2 application operates independently of the GUI code. You can run it with `python application/florence-2-vision/florence2_app.py` inside the Florence-2 Docker container. For GUI integration, the `set_parameters()` method was added to the `FlorenceApp` class. This method updates two fields in the Florence-2 operator:
+
+```python
+class FlorenceApp(Application):
+    def set_parameters(self, task, prompt):
+        """Set parameters for the Florence2Operator."""
+        if self.florence_op:
+            self.florence_op.task = task
+            self.florence_op.prompt = prompt
+```
+
+These updated parameters are passed to the model during the next `compute()` method execution of the Florence-2 operator.
+
+### GUI Code
+
+The GUI code resides in [qt_app.py](https://github.com/nvidia-holoscan/holohub/blob/main/applications/florence-2-vision/qt_app.py).
+The code in this file defines a class for the main window which calls `setupUi()` and `runHoloscanApp()` when the instance is initialized.
+
+```python
+class Window(QMainWindow):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setupUi()  # Setup the UI
+        self.runHoloscanApp()  # Run the Holoscan application
```

Comment:
Does this run in the same process? PySide handles UI threading?

Reply:
Yes, this runs in the same process.  PySide has a QThread construct for threads